app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -153,9 +162,44 @@ class BrowserToolbarIntegration(
         toolbar.display.setUrlBackground(urlBackground)
     }
 
+    private fun setActionButtons(context: Context) {
+        if (store.state.findCustomTabOrSelectedTab(customTabId)?.isCustomTab() == false) {
+            val eraseAction = BrowserToolbar.Button(
+                imageDrawable = AppCompatResources.getDrawable(
+                    context,
+                    R.drawable.mozac_ic_delete
+                )!!,
+                contentDescription = context.getString(R.string.content_description_erase),
+                iconTintColorResource = R.color.primaryText,
+                listener = {
+                    val openedTabs = store.state.tabs.size
+                    TabCount.eraseButtonTapped.record(TabCount.EraseButtonTappedExtra(openedTabs))
+                    eraseActionListener.invoke()
+                }
+            )
+            toolbar.addNavigationAction(eraseAction)
+
+            store.flowScoped { flow ->
```

Comment:
This method returns a `CoroutineScope`. We should make sure to keep a reference to it and cancel it (at the latest from `stop()`).